lib/src/util/number.dart

```diff
@@ -112,9 +112,141 @@ double fuzzyAssertRange(double number, int min, int max, [String? name]) {
+SassNumber tan(SassNumber number) {
+  return SassNumber(math
+      .tan(number.assertNumber("number").coerceValueToUnit("rad", "number")));
+}
+
+/// Returns the arctangent of [number].
+SassNumber atan(SassNumber number) {
+  return SassNumber.withUnits(math.atan(number.value) * 180 / math.pi,
+      numeratorUnits: ['deg']);
+}
+
+/// Returns the arcsine of [number].
+SassNumber asin(SassNumber number) {
+  return SassNumber.withUnits(math.asin(number.value) * 180 / math.pi,
+      numeratorUnits: ['deg']);
+}
+
+/// Returns the arccosine of [number]
+SassNumber acos(SassNumber number) {
+  return SassNumber.withUnits(math.acos(number.value) * 180 / math.pi,
+      numeratorUnits: ['deg']);
+}
+
+/// Returns the absolute value of [number].
+SassNumber abs(SassNumber number) {
+  return SassNumber(number.value.abs());
+}
+
+/// Returns the logarithm of [number] with respect to [base].
+SassNumber log(SassNumber number, SassNumber? base) {
+  if (base != null) {
+    return SassNumber(math.log(number.value) / math.log(base.value));
+  }
+  return SassNumber(math.log(number.value));
+}
+
+/// Returns the value of [num1] raised to the power of [num2].
+SassNumber pow(SassNumber num1, SassNumber num2) {
+  num1.assertNoUnits();
+  num2.assertNoUnits();
+  return SassNumber(math.pow(num1.value, num2.value));
+}
+
+/// Returns the arctangent for [y] and [x].
+SassNumber atan2(SassNumber y, SassNumber x) {
+  return SassNumber.withUnits(
+      math.atan2(y.value, x.convertValueToMatch(y, 'x', 'y')) * 180 / math.pi,
+      numeratorUnits: ['deg']);
+}
+
+/// Returns a rounded [number] based on a selected rounding [strategy],
+/// to the nearest integer multiple of [step].
+SassNumber roundWithStep(
+    SassString strategy, SassNumber number, SassNumber step) {
+  if (step.value == 0) return SassNumber(double.nan).coerceToMatch(number);
+  if (number.value.isInfinite ||
+      step.value == 0 ||
+      number.value.isNaN ||
+      step.value.isNaN) {
+    if (!step.value.isInfinite) {
+      return number;
+    }
+    return SassNumber(double.nan).coerceToMatch(number);
+  }
+
+  if (step.value.isInfinite) {
+    if (number.value == -0.0) return number;
+    if (strategy.text == "nearest" || strategy.text == "to-zero") {
+      return number.value >= 0
+          ? SassNumber(0).coerceToMatch(number)
+          : SassNumber(-0.0).coerceToMatch(number);
+    }
+    if (strategy.text == "up") {
+      return number.value >= 0
+          ? number.value == 0
+              ? number
+              : SassNumber(double.infinity).coerceToMatch(number)
+          : SassNumber(-0.0).coerceToMatch(number);
+    }
+    if (number.value < 0 && strategy.text == "down") {
+      return number.value <= -0.0
+          ? number.value == -0.0
+              ? number
+              : SassNumber(-double.infinity).coerceToMatch(number)
+          : SassNumber(0).coerceToMatch(number);
+    }
+
+    return SassNumber(0).coerceToMatch(number);
```

Comment:
Oh right, I though I had seen `if (number.value == -0.0) return -0` in the doc.  Although for the down case the spec mentions: `If strategy's value is "down", return negative infinity if number's value is negative, -0 if number's value is -0, and +0 otherwise.` which is why I'm doing in this line. I think it does changes behavior because the tests have different results when we do `math.div(1, -0)` or `math.div(1, 0)`. But yes, a switch statement sounds like a better idea.